File: domain/src/main/java/Saga.java
import java.util.HashMap;
import java.util.concurrent.Future;
import java.util.function.Consumer;

public interface Saga {
    String getId();

    Integer getVersion();

    void Transition(Object message);

    Iterable<Object> getUncommittedEvents();

    void clearUncommittedEvents();

    Iterable<Object> getUndispatchedMessages();

    void clearUndispatchedMessages();
}

File: modules/domain/src/main/kotlin/org/openmedstack/domain/Saga.kt
package org.openmedstack.domain

import org.openmedstack.events.BaseEvent

interface Saga {
    val id: String
    val version: Int
    @Throws(HandlerForDomainEventNotFoundException::class)
    fun transition(message: BaseEvent)
    val uncommittedEvents: Iterable<BaseEvent>
    fun clearUncommittedEvents()
    val undispatchedMessages: Iterable<Any>
    fun clearUndispatchedMessages()
}
